### PR TITLE
Filtering spaces

### PR DESCRIPTION
**Pull request by chriswhitehouse**
#18

### CODE REVIEW

File: spec/features/filtering_spaces_by_date_spec.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
+feature 'So i can see spaces in a specified date range' do
+  scenario 'I would like to be able to filter the list by date of availibility' do
+
+    Space.create(
+      name: "Test name",
+      description: "Test description",
+      date_available_from: "2021-02-02",
+      date_available_to: "2021-02-20",
+      price: "50"
+      )
```
Comment (PiperS52) [inline spec/features/filtering_spaces_by_date_spec.rb]: user_id

File: spec/features/filtering_spaces_by_date_spec.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+      Space.create(
+        name: "Another name",
+        description: "Another Test description",
+        date_available_from: "2021-03-09",
+        date_available_to: "2021-03-29",
+        price: "50"
+        )
```
Comment (PiperS52) [inline spec/features/filtering_spaces_by_date_spec.rb]: user_id

File: spec/features/filtering_spaces_by_date_spec.rb [hunk 3]
```diff
@@ -0,0 +1,33 @@
+feature 'So i can see spaces in a specified date range' do
+  scenario 'I would like to be able to filter the list by date of availibility' do
+
```
Comment (PiperS52) [inline spec/features/filtering_spaces_by_date_spec.rb:3]: user = User.create()

File: spec/features/filtering_spaces_by_date_spec.rb (hunk 1 above)
Comment (chriswhitehouse) [reply spec/features/filtering_spaces_by_date_spec.rb]: updated

File: spec/units/space_spec.rb [hunk 4]
```diff
@@ -65,4 +65,29 @@
...
 
+  describe ".filter" do
+    it "should only list spaces available within required date range" do
+      Space.create(
+        name: "Test name",
+        description: "Test description",
+        date_available_from: "2021-02-01",
+        date_available_to: "2021-02-20",
+        price: "50"
+        )
```
Comment (PiperS52) [inline spec/units/space_spec.rb]: same as above

**Review by PiperS52 — CHANGES_REQUESTED** [review]
Please make changes

**Review by PiperS52 — APPROVED** [review]
looks good

**Comment by chriswhitehouse** [comment]
[As a user So i can see spaces in a specified date range I would like to be able to filter the list by date of availibility.](https://trello.com/c/qhvRsHBj/8-as-a-user-so-i-can-see-spaces-in-a-specified-date-range-i-would-like-to-be-able-to-filter-the-list-by-date-of-availibility)

**Pull request merged**